californium-core/src/main/java/org/eclipse/californium/core/network/Matcher.java

```diff
@@ -173,6 +177,16 @@ public void sendRequest(Exchange exchange, Request request) {
 		
 		exchangesByMID.put(idByMID, exchange);
 		exchangesByToken.put(idByToken, exchange);
+		
+		if(request.isMulticast()) {
+			executor.schedule(new Runnable() {
+				@Override
+				public void run() {
+					exchange.getCurrentRequest().setTimedOut(true);
+					exchange.setTimedOut();
```

Comment:
Timeout seems to be related to CON messages. 
According RFC 7252, 8.1, page 65, all multicast messages MUST be NON so a timeout would not apply.
Try cancel()?

Reply:
Yes, I will try it.

Reply:
I would go for cancel and complete.

Reply:
More advanced:
check the exchange, if it has already at least one response. If so, just complete the exchange. If not cancel the request and complete the exchange. I think, this would be the best approach to hande the multiple responses.

Reply:
I implemented your two last recommendation and it works fine for me.